x-pack/plugins/infra/public/pages/metrics/inventory_view/components/node_details/tabs/metrics/metrics.tsx

```diff
@@ -0,0 +1,471 @@
+    ],
+    [],
+    nodeType,
+    sourceId,
+    currentTime,
+    accountId,
+    region,
+    false,
+    {
+      interval: '1m',
+      to: currentTime,
+      from: currentTime - time,
+      ignoreLookback: true,
+    }
+  );
+
+  const getDomain = useCallback(
+    (timeseries: MetricsExplorerSeries, ms: MetricsExplorerOptionsMetric[]) => {
+      const dataDomain = timeseries ? calculateDomain(timeseries, ms, false) : null;
+      return dataDomain
+        ? {
+            max: dataDomain.max * 1.1, // add 10% headroom.
+            min: dataDomain.min,
+          }
+        : { max: 0, min: 0 };
+    },
+    []
+  );
+
+  const dateFormatter = useCallback((timeseries: MetricsExplorerSeries) => {
+    if (!timeseries) return () => '';
+    const firstTimestamp = first(timeseries.rows)?.timestamp;
+    const lastTimestamp = last(timeseries.rows)?.timestamp;
+
+    if (firstTimestamp == null || lastTimestamp == null) {
+      return (value: number) => `${value}`;
+    }
+
+    return niceTimeFormatter([firstTimestamp, lastTimestamp]);
+  }, []);
+
+  const networkFormatter = useMemo(() => createInventoryMetricFormatter({ type: 'rx' }), []);
+  const cpuFormatter = useMemo(() => createInventoryMetricFormatter({ type: 'cpu' }), []);
+  const memoryFormatter = useMemo(
+    () => createInventoryMetricFormatter({ type: 's3BucketSize' }),
+    []
+  );
+  const loadFormatter = useMemo(() => createInventoryMetricFormatter({ type: 'load' }), []);
+
+  const mergeTimeseries = useCallback((...series: MetricsExplorerSeries[]) => {
+    const base = series[0];
+    const otherSeries = series.slice(1);
+    base.rows = base.rows.map((b, rowIdx) => {
+      const newRow = { ...b };
+      otherSeries.forEach((o, idx) => {
+        newRow[`metric_${idx + 1}`] = o.rows[rowIdx].metric_0;
+      });
+      return newRow;
+    });
+    return base;
+  }, []);
+
+  const buildChartMetricLabels = useCallback(
+    (labels: string[], aggregation: MetricsExplorerAggregation) => {
+      const baseMetric = {
+        color: Color.color0,
+        aggregation,
+        label: 'System',
+      };
+
+      return labels.map((label, idx) => {
+        return { ...baseMetric, color: Color[`color${idx}` as Color], label };
+      });
+    },
+    []
+  );
+
+  const pointerUpdate = useCallback(
+    (event: PointerEvent) => {
+      chartRefs.forEach((ref) => {
+        if (ref.current) {
+          ref.current.dispatchExternalPointerEvent(event);
+        }
+      });
+    },
+    [chartRefs]
+  );
+
+  const isDarkMode = useUiSetting<boolean>('theme:darkMode');
+  const tooltipProps = {
+    headerFormatter: (tooltipValue: TooltipValue) =>
+      moment(tooltipValue.value).format('Y-MM-DD HH:mm:ss.SSS'),
+  };
+
+  const getTimeseries = useCallback(
+    (metricName: string) => {
+      if (!nodes || !nodes.length) {
+        return null;
+      }
+      return nodes[0].metrics.find((m) => m.name === metricName)!.timeseries!;
```

Comment:
I think we've started using `first` from `lodash` instead of accessing `Array[0]`

Reply:
What's the advantage of using first? Happy to add it if it makes more sense.

Reply:
It's more of an advantage using `last` instead of `Array[Array.length - 1]`, so using `first` helps with consistency there.